src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        }
+        if (type.equals(Track.class)) {
+            return new ArrayList<>(new TrackProducer<>().get());
+        }
+        if (type.equals(Excavator.class)) {
+            return new ArrayList<>(new ExcavatorProducer<>().get());
+        }
+        throw new ClassCastException("Wrong class");
```

Comment:
To my mind better to throw  IllegalArgumentException()

Reply:
Agree, done.